### ISSUE TITLE
Call to non configured method "ToString" of strict fake.

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#639]
This test:

```
[Test]
void Test()
{
    var foo = A.Fake<IFoo>();
    var baz = A.Fake<IBaz>(o => o.Strict());
    foo.Bar(baz);
    A.CallTo(() => foo.Bar(baz)).MustHaveHappened();
}

public interface IFoo
{
    void Bar(IBaz baz);
}

public interface IBaz
{
    
}
```

throws an error:

```
Call to non configured method "ToString" of strict fake.
Stacktrace:
   at FakeItEasy.FakeOptionsExtensions.<Strict>b__0[T](IFakeObjectCall call)
   at FakeItEasy.Configuration.BuildableCallRule.Apply(IInterceptedFakeObjectCall fakeObjectCall)
   at FakeItEasy.Core.FakeManager.ApplyRule(CallRuleMetadata rule, IInterceptedFakeObjectCall fakeObjectCall)
   at FakeItEasy.Core.FakeManager.FakeItEasy.Core.IFakeCallProcessor.Process(IWritableFakeObjectCall fakeObjectCall)
   at FakeItEasy.Creation.CastleDynamicProxy.CastleDynamicProxyGenerator.ProxyInterceptor.Intercept(IInvocation invocation)
   at Castle.DynamicProxy.AbstractInvocation.Proceed()
   at Castle.Proxies.ObjectProxy_1.ToString()
   at FakeItEasy.Expressions.ArgumentConstraints.EqualityArgumentConstraint.ToString()
   at FakeItEasy.Expressions.ArgumentConstraints.EqualityArgumentConstraint.WriteDescription(IOutputWriter writer)
   at FakeItEasy.Expressions.ExpressionCallMatcher.AppendArgumentsListString(StringBuilder result)
   at FakeItEasy.Expressions.ExpressionCallMatcher.ToString()
   at FakeItEasy.Expressions.ExpressionCallMatcher.get_DescriptionOfMatchingCall()
   at FakeItEasy.Expressions.ExpressionCallRule.get_DescriptionOfValidCall()
   at FakeItEasy.Configuration.BuildableCallRule.WriteDescriptionOfValidCall(IOutputWriter writer)
   at FakeItEasy.Configuration.RuleBuilder.MustHaveHappened(Repeated repeatConstraint)
   at FakeItEasy.AssertConfigurationExtensions.MustHaveHappened(IAssertConfiguration configuration)
   at UserQuery.Main()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

But there is no call to `ToString()` in the user code... it's FakeItEasy that calls it (even though it's not really necessary, since the call did happen). FakeItEasy probably shouldn't try to call `ToString()` on a strict fake (or perhaps there should be a default `ToString()` implementation, even on strict fakes)

**Comment by adamralph** [comment]
:+1: I can reproduce.

**Comment by blairconrad** [comment]
Duplicate of #457?

**Comment by blairconrad** [comment]
> Duplicate of #457?

Let me rephrase.

"Duplicate of #457."
Closing, unless someone rushes in with a reason not to.

**Comment by adamralph** [comment]
> Duplicate of #457?

Indeed.

**Comment by thomaslevesque** [comment]
Ah, yes, didn't see that one.
